05/app.ts

```diff
@@ -0,0 +1,91 @@
+abstract class Character {
+    private _name: string;
+    private _health: number;
+    public burnEffect: boolean = false; 
+
+    constructor(name: string, health: number = 100) {
+        this._name = name;
+        this.health = health;
+    }
+
+    set health(value: number) {
+        if (value <= 0) {
+            this._health = 0;
+            console.log(`${this.name} umiera`);
+        } else if (value > 100) {
+            this._health = 100;
+        } else {
+            this._health = value;
+        }
+    }
+
+    get health() {
+        return this._health;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    abstract useSkill(target: Character): void;
+
+    static performAction(actor: Character, target: Character) {
+        actor.useSkill(target);
+    }
+
+    applyBurnIfNeeded() {
```

Comment:
Czy to faktycznie musi być taka nazwa. Raczej starałbym się to zrobić bardziej elastycznie. Inna postać może mieć efekt "trucizny". Raczej chodzi o to, że można mieć efekt tracenia życia w jakimś konkretnym przypadku, a nie tylko podczas "spalenia". Tak tylko zwracam uwagę, że warto bardziej elastycznie podejść do takiego rozwiazania. Zwróciłbym też uwagę, że zawsze mamy ten element powiązany z useSkill więc może warto "w rodzicu" uruchamiać tę metodą